### DISCUSSION TITLE
config files, files-directory structure for customization

### DISCUSSION THREAD

**Discussion by techopus** [TerriaJS/terriajs#6505] [Q&A]
I am working on to customize it for moon(in place of earth). For instance I need to change the size/shape to moon, I go to node_modules and find each related .js files and change the size object, which means I need to hard code everything. I am not sure if this is the way to do but it doesn't give accurate result, reason maybe that there are other files which needs to be updated but I am not able to know which and where and how many. So, is there proper way or structure of files or any parent/config files where only we can edit and add functions and should be enough. Maybe what I am doing is the only way but I just wanted to be clear. I hope the question is clear enough. Please do help me with clear ideas on this. @nf-s Maybe you can help me out with this?

**Comment by tephenavies** [comment]
Hi @techopus. TerriaJS uses Cesium to draw the 3d model of the Earth. I'd suggest getting rendering of the moon working in Cesium first (e.g. in Cesium Sandcastle) and then modify parts of https://github.com/TerriaJS/terriajs/blob/main/lib/Models/Cesium.ts (within TerriaJS) to use your moon rendering in TerriaJS.

**Reply by techopus** [reply]
@steve9164 Thanks a lot. I think that's what i needed to know.
